test/functional/architectures_controller_test.rb

```diff
@@ -64,6 +64,13 @@ def test_destroy
     assert !Architecture.exists?(architecture.id)
   end
 
+  test "403 response contains missing permissions" do
+    setup_user
+    get :edit, { :id => Architecture.first.id },  {:user => users(:one).id, :expires_at => 5.minutes.from_now}
```

Comment:
Same as for the API, `application_controller_subclass_test` seems like a better fit for this kind of tests?